### PR TITLE
refactor: query library and reporting

### PR DESCRIPTION
**Pull request by elad-pticha**

### CODE REVIEW

File: src/cmdline.py [hunk 1]
```diff
@@ -175,24 +175,47 @@ def raven() -> None:
...
         help="Generate report from indexed Actions - Beta Version",
     )
+
+    report_parser.add_argument(
+        "--tag",
+        "-t",
+        action="append",
+        type=str,
+        default=[],
+        help="Filter detections with spesific tag",
```
Comment (alex-ilgayev) [inline src/cmdline.py:185]:
```suggestion
        help="Filter detections with specific tag",
```

File: src/cmdline.py [hunk 2]
```diff
@@ -175,24 +175,47 @@ def raven() -> None:
...
-        "--slack",
+        "--severity",
         "-s",
-        action="store_const",
-        default=REPORT_SLACK_DEFAULT,
-        const=True,
-        help=f"Send report to slack channel, default: {REPORT_SLACK_DEFAULT}",
+        type=str,
+        default=[],
+        help="Filter detections with spesific severity",
```
Comment (alex-ilgayev) [inline src/cmdline.py:192]:
```suggestion
        help="Filter detections with specific severity",
```

File: src/cmdline.py [hunk 3]
```diff
@@ -175,24 +175,47 @@ def raven() -> None:
...
+        "--tag",
+        "-t",
+        action="append",
+        type=str,
+        default=[],
+        help="Filter detections with spesific tag",
+    )
     report_parser.add_argument(
-        "--slack",
+        "--severity",
```
Comment (alex-ilgayev) [inline src/cmdline.py:188]: Maybe do it minimum severity instead?

File: src/detections/__init__.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+from src.config.config import Config
+
+
+class Detection(object):
```
Comment (alex-ilgayev) [inline src/detections/__init__.py:4]: This isn't clear.
Is it a detection? or is it a query/policy?
I think better to separate it.

Create a class named Query, that will be loaded from the `yml` files in `library`. and then you'll have list of queries.
On top of that you can create query library. from there can invoke the the queries.

I think it is important to build it right so we'll be able to scale it.

**Review by alex-ilgayev — CHANGES_REQUESTED** [review]

File: src/storage/neo4j_graph.py [hunk 5]
```diff
@@ -3,7 +3,8 @@
 from py2neo.data import Node
 from typing import Tuple, Optional
 import src.logger.log as log
-from src.library import query_library
+
+# from src.library import query_library
```
Comment (oreenlivnicode) [inline src/storage/neo4j_graph.py:7]: Delete?

**Review by oreenlivnicode — COMMENTED** [review]
Nice

File: src/detections/__init__.py (hunk 4 above)
Comment (elad-pticha) [reply src/detections/__init__.py:4]: Changed to Query. Each query will have a `run` method to return vulnerable workflow paths.

**Review by alex-ilgayev — APPROVED** [review]

**Pull request merged**